treefmt.nix

```diff
@@ -0,0 +1,32 @@
+{ lib, pkgs, ... }:
+
+{
+  tree-root-file = ".git/config";
```

Comment:
Setting a tree-root-file is no longer needed; the default treefmt behaviour is to ask `git rev-parse --show-toplevel`.

If you do specify a tree-root-file, `.git/config` will break for git worktrees, where `.git` is a file. On the flip side, `.git` would break for primary-worktrees since the `tree-root-file` cannot be a directory.

So if you do sepcify a file, it's better to choose a known file from this repo, known to only be at the toplevel, such as `flake.lock` or `.github/workflows/check.yml`.